### ISSUE TITLE
Command mode supports positional arguments

### ISSUE THREAD

**Issue by klesh** [nushell/nushell#11897]
### Related problem

For POSIX-compliant shell, we can do this
```
sh -c 'echo "$@"' -- foo bar
```
would print
```
foo bar
```

However, there is no way to pass arguments in the same manner to the command with `nushell` which causes some usability problems, check https://github.com/gokcehan/lf/pull/1606 for more detail

### Describe the solution you'd like

Support `--` and maybe a new magic variable like `argv`?
```
nu -c 'echo ...$argv' -- foo bar
```

### Additional context and details

I believe the use case from https://github.com/gokcehan/lf/pull/1606#issuecomment-1951736746 is legit and supporting it would help `nushell` gain more momentum.

**Comment by NotTheDr01ds** [comment]
Ran into some similar problems with `fzf` passing a preview command-line to `nu`.  In that case, it was differences in quoting between POSIX/Fish vs. Nushell.

I think there's a case to be made for providing some commonality with other shells in the way Nushell is *invoked* at least.

**Comment by klesh** [comment]
@fdncred Would you like to take a look at the issue and share your thoughts? I would like to work on it if we are on the same page.

**Comment by fdncred** [comment]
The last time we talked about this I thought that `def --wrapped` custom commands would suffice. I could be remembering wrong though. For what it's worth, I'd probably support something like `nu -c 'echo $argv' -- foo bar` because it's pretty widely used, although others may not.

Also, saying posix shells do x is not a real motivating factor for nushell to do anything since we have never planned on being posix compliant. However, we do want nushell to feel comfortable and "at home" to a certain degree, but that definitely doesn't mean implementing everything a posix shell does. Sometimes keeping muscle memory is good. Other times we should just learn something new.

This is how `def --wrapped` works. It allows you to pass most anything through
```nushell
❯ def --wrapped test [...args] { print ...$args }
❯ test -- a b c
--
a
b
c
```

**Comment by klesh** [comment]
@fdncred Sure, I didn't mean that `nushell` should obey the posix standard, I believe it is too old and should have been renewed a long time ago.
My feeling is more like: look, even good old Posix shell has it covered 😄, and `nushell` doesn't seem to be supporting such a use case at all.

I may not understand the `def --wrapped` correctly, but it doesn't seem to be working:
![image](https://github.com/nushell/nushell/assets/61080/b837f71b-00df-4a79-96f0-903f11489518)

So, I think we all agree that it is a good thing to have, I will be working on the feature maybe next week, it could take a while since I am new to the project and Rust 😄

**Comment by fdncred** [comment]
You're trying to pass `-- foo bar` to `nu`, not the `tmp` custom command which is fine, but my example was for custom commands supporting those flags, not the nushell executable. The nushell executable does not support it and it would have to be added via rust code.

**Comment by klesh** [comment]
@fdncred Yes, you are correct. I just wanted to make sure we are on the same page. I would try to add the support to the double dash `--` for the `nu` executable, and the temp custom command example you showed is a good pointer for the implementation.

**Comment by fdncred** [comment]
Ok, as long as it doesn't interfere with params being passed to scripts. I'm pretty sure this is allowed now like `nu script.nu param1 param2` where param1 and param2 are passed into script.nu as parameters. I could be remembering wrong though. Seems like it wouldn't interfere, but we just don't want to break existing functionality.

**Comment by ian-h-chamberlain** [comment]
Hey @fdncred, this issue's been open a while now but I wonder if it makes sense to revisit now with #18299 implemented.

> For what it's worth, I'd probably support something like `nu -c 'echo $argv' -- foo bar` because it's pretty widely used, although others may not.

Are you still in favor of having some functionality like this, if I were to try and make a PR implementing it? 

With `--` handling now on commands, maybe `nu -c` should just forward all args verbatim to the `-c` script and let the commands handle? e.g. 
```console
$ nu --version; cat main.nu;  nu main.nu -- foo bar baz; cat main-wrapped.nu ; nu main-wrapped.nu -- foo bar baz
0.114.0
─────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
     │ File: main.nu
─────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ def main [...rest] {
   2 │   $rest | print
   3 │ }
─────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
╭───┬─────╮
│ 0 │ foo │
│ 1 │ bar │
│ 2 │ baz │
╰───┴─────╯
─────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
     │ File: main-wrapped.nu
─────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ def --wrapped main [...rest] {
   2 │   $rest | print
   3 │ }
─────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
╭───┬─────╮
│ 0 │ --  │
│ 1 │ foo │
│ 2 │ bar │
│ 3 │ baz │
╰───┴─────╯
```

Then we'd have something like 
```console
$ nu -c 'def main [...rest] { $rest | print }' -- foo bar baz 
╭───┬─────╮
│ 0 │ foo │
│ 1 │ bar │
│ 2 │ baz │
╰───┴─────╯
$ nu -c 'def --wrapped main [...rest] { $rest | print }' -- foo bar baz
╭───┬─────╮
│ 0 │ --  │
│ 1 │ foo │
│ 2 │ bar │
│ 3 │ baz │
╰───┴─────╯
```

Or were you thinking about actually having a special variable to avoid the `def main` boilerplate entirely?

**Comment by fdncred** [comment]
> Are you still in favor of having some functionality like this

Yes

> With -- handling now on commands, maybe nu -c should just forward all args verbatim to the -c script and let the commands handle?

Ok

> Or were you thinking about actually having a special variable to avoid the def main boilerplate entirely?

Nope, wasn't thinking of a special variable. -- just wasn't added to the nu executable yet.
